### PR TITLE
crypto: use recoverable signature for secp256k1

### PR DESCRIPTION
**Pull request by joyqvq**
https://github.com/MystenLabs/sui/issues/3270

### CODE REVIEW

File: network/src/primary.rs [hunk 1]
```diff
@@ -72,6 +72,12 @@ impl PrimaryNetwork {
         self.send_message(address, message).await
     }
 
+    // Safety
```
Comment (kchalkias) [inline network/src/primary.rs:75]: We don't need in this PR

File: crypto/src/secp256k1.rs [hunk 2]
```diff
@@ -29,11 +29,14 @@ pub struct Secp256k1PrivateKey {
     pub bytes: OnceCell<[u8; constants::SECRET_KEY_SIZE]>,
 }
 
+// Compact signature followed by one extra byte for recover id, used to recover public key from signature.
+pub const RECOVERABLE_SIGNATURE_SIZE: usize = constants::COMPACT_SIGNATURE_SIZE + 1;
```
Comment (kchalkias) [inline crypto/src/secp256k1.rs:33]: thus, 65 bytes! great

**Review by kchalkias — COMMENTED** [review]
The title of the PR does not match functionality (expecting non-malleable sigs (with reduced s value - but see ecrecover. 
Can we update the title to reflect that?

**Review by kchalkias — DISMISSED** [review]
LGTM

**Review by kchalkias — COMMENTED** [review]
We need to rebase and fix any failing test

**Review by kchalkias — APPROVED** [review]
Thanks @joyqvq. This PR is in a mergeable state now. We might revisit the default hash function (and only use keccak in our tests).

File: crypto/Cargo.toml [hunk 3]
```diff
@@ -20,7 +20,9 @@ tokio = { version = "1.20.0", features = ["sync", "rt", "macros"] }
 ark-bls12-377 = { version = "0.3.0", features = ["std"], optional = true }
 hkdf = { version = "0.12.3", features = ["std"] }
 serde_with = "1.14.0"
-rust_secp256k1 = { version = "0.24.0", package = "secp256k1", features = ["rand", "bitcoin_hashes"] }
+serde_json = "1.0.80"
+schemars ="0.8.10"
```
Comment (huitseeker) [inline crypto/Cargo.toml:24]: Small merge error on this one

**Pull request merged**